File: src/main/java/compiler/parser/CommentParser.java
package compiler.parser;

import compiler.block.Block;
import compiler.block.comments.CommentBlock;
import compiler.tokenizer.Tokenizer;

public class CommentParser extends Parser<CommentBlock> {

	@Override
	public boolean shouldParse(String line) {

		return line.matches("//.*");
	}



	@Override
	public CommentBlock parse(Block superBlock, Tokenizer tokenizer) {

		return new CommentBlock(superBlock);
	}
}

File: src/main/scala/test_classes/parser/comments/CommentParser.scala
package test_classes.parser.comments

import test_classes.block.Block
import test_classes.block.comments.CommentBlock
import test_classes.parser.Parser
import test_classes.tokenizer.Tokenizer

class CommentParser extends Parser[CommentBlock] {

  def shouldParse(line: String): Boolean = line.matches("//.*")

  def parse(superBlock: Block, tokenizer: Tokenizer): CommentBlock = new CommentBlock(superBlock)

}
